Sources/Datadog/DDRUMMonitor.swift

```diff
@@ -11,7 +11,18 @@ import UIKit
 /// `DDRUMMonitor` allows you to record User events that can be explored and analyzed in Datadog Dashboards.
 /// You can only have one active `RUMMonitor`, and should register/retrieve it from the `Global` object.
 public class DDRUMMonitor {
-    // MARK: - Public
+    // MARK: - Public properties
+
+    /// Global attributes to be injected into every command
+    public var attributes: [AttributeKey: AttributeValue] { [:] }
+
+    /// Sets a global attribute to be injected into every command
+    /// - Parameters:
+    ///   - key: key of the global attribute. If key already exists, new value replaces old value
+    ///   - value: attribute value. If nil, key is removed from global attributes
+    public func setAttribute(forKey key: AttributeKey, value: AttributeValue?) { }
```

Comment:
`var attributes` is used in `RUMMonitor.aggregate` method, it also serves as thread-safe public getter of private property

i looked at Logger API but i didn't understand 
1. why we hide user's attributes from the user, why can't they get what they set?
2. why don't we have one single `set(attribute: Value?)` to add/update/remove attributes?

a similar request was made yesterday via Slack (i've just sent you the link)

Reply:
```swift
public var attributes: [AttributeKey: AttributeValue]
```
is basically extending the public interface with the interface of `Dictionary`. If we decide to change from storing attributes as dictionary we will be put into trouble.

> why we hide user's attributes from the user, why can't they get what they set?

For the sake of keeping things simple. If people don't need it, we don't add it (as we don't have it for `Logger`).

> why don't we have one single set(attribute: Value?) to add/update/remove attributes?

Because it's less explicit and people need to guess that setting `nil` will remove the attribute. This is cleaner and sets the smaller API comments in the right places:
```swift
public func addAttribute(forKey key: AttributeKey, value: AttributeValue)
public func removeAttribute(forKey key: AttributeKey)
```
It's also how people used to work with `Datadog.Logger`.

Reply:
ℹ️  in short, okay i'm gonna copy `Logger` API to `RUM` because i need this feature for my main ticket asap ✅ 
i'm resolving the conversation as i'll make changes and the discussion seems to be around beliefs and simplicity

---

but i'd like to leave my opinion here for the records 🎗️ 

the new type must support `init(from: Dictionary)` as we receive attributes as `Dictionary` in public methods.
if this type doesn't support conversion back to a `Dictionary`, yes then we have a problem. if you genuinely believe that may happen, then it's fine 👌 
we can do smth like `var attributes: Collection` too btw.?

if a user needs an attribute value, they need to double their sources of truth; honestly, i don't think this is simple.
also, we received a similar request about getting set values just yesterday.

`set(value: nil, for: key)` has other examples in Swift, `Dictionary.subscript` is pretty similar and common too.
if i were a user dealing with an optional value, i would need to do
```swift
if let someValue = myOptionalValue {
  rum.addAttribute(forKey: myKey, value: someValue)
} else {
  rum.removeAttribute(forKey: myKey)
}
```
vs. `rum.set(myOptionalValue, for: myKey)`
either way works to be honest 👌  i just don't understand how we can go with the former while we are for simplicity 😕